File: src/main/scala/org/biodatagraphdb/alsdb/value/Uniprot2Reactome.scala
package org.biodatagraphdb.alsdb.value

import org.apache.commons.csv.CSVRecord

case class Uniprot2Reactome(uniProtId:String,
                            reactomeId:String,
                            url:String, eventName:String,
                            evidenceCode:String, species:String
                           ){

}

object Uniprot2Reactome extends ValueTrait {

  def parseCSVRecord (record:CSVRecord): Uniprot2Reactome= {
    new Uniprot2Reactome(
      record.get("UniProt_ID"), record.get("Reactome_ID"),
      record.get("URL"), record.get("Event_Name"),
      record.get("Evidence_Code"), record.get("Species")
    )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/Uniprot2Reactome.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class Uniprot2Reactome(
        val uniProtId: String,
        val reactomeId: String,
        val url: String,
        val eventName: String,
        val evidenceCode: String,
        val species: String
) {
    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): Uniprot2Reactome =
                Uniprot2Reactome(
                        record.get("UniProt_ID"),
                        record.get("Reactome_ID"),
                        record.get("URL"),
                        record.get("Event_Name"),
                        record.get("Evidence_Code"),
                        record.get("Species")
                )
    }
}